src/semaphore.c

```diff
@@ -630,6 +680,11 @@ _dispatch_group_wait_slow(dispatch_semaphore_t dsema, dispatch_time_t timeout)
 			ret = sem_wait(&dsema->dsema_sem);
 		} while (ret == -1 && errno == EINTR);
 		DISPATCH_SEMAPHORE_VERIFY_RET(ret);
+#elif USE_FUTEX_SEM
+		do {
+			ret = _dispatch_futex_wait(&dsema->dsema_futex, NULL);
+		} while (ret == -1 && errno == EINTR);
```

Comment:
you wrote that EINTR loop 3 or 4 times already, _dispatch_futex_wait should handle EINTR itself.